test/e2e/apps/daemon_set.go

```diff
@@ -62,7 +62,7 @@ var NamespaceNodeSelectors = []string{"scheduler.alpha.kubernetes.io/node-select
 // happen.  In the future, running in parallel may work if we have an eviction
 // model which lets the DS controller kick out other pods to make room.
 // See http://issues.k8s.io/21767 for more details
-var _ = SIGDescribe("Daemon set [Serial]", func() {
+var _ = SIGDescribe("Daemon set", func() {
```

Comment:
I'm not convinced we're OK to remove this tag.

I don't think the [Serial] tag was placed here because of node labels.  The comment right above this `SIGDescribe` explains why the Serial tag was added, and it's not clear to me that concern has been addressed?

Reply:
Looking into it Aaron @spiffxp

Reply:
Pod preemption is handled by the default scheduler. When preemption is enabled, the daemonSet controller makes scheduling decisions with no consideration for pod priority and preemption. It is not desired by 'kubelet' to evict a DaemonSet pod as the pod is immediately recreated and rescheduled back to the same node. For now, kubelet has not been provided the ability to distinguish between a pod created by DaemonSet versus any other object. So Kubelet treats even the daemon pods as normal pods. As a result, kubelet cannot filter the DaemonSet pods from the candidate set of pods provided to the eviction strategy. In an ideal scenario, DaemonSet pods should have a higher priority and should be rejected for eviction strategy.

As per the discussion in https://groups.google.com/forum/#!topic/kubernetes-sig-scheduling/kMG7yfONwY4, the conclusion was to keep the current behaviour of daemonSet controller and let it schedule the DaemonSet pods. Also, there is no synchronisation between DaemonSet Controller and the default scheduler which can lead to a race condition. Due to lack of an eviction strategy, the race condition cannot be resolved in DC's favour.

Hence, the request for removal of SERIAL tag from DaemonSet conformance test can be put on hold until a priority mechanism to distinguish between a regular pod and the daemon pods is implemented allowing the Daemon pods to be rejected from eviction strategy. Closely follows #22212

Conclusion: I propose to keep the DaemonSet tests in [SERIAL] and hence revert the changes made with respect to his PR.